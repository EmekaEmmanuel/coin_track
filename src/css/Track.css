* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  font-family: 'Montserrat', sans-serif;
  letter-spacing: 0.2px;
}

.app_container {
  margin: auto;
  background-color: #111315;
}

.main_tag {
  text-align: center;
}

.home_section1 {
  padding: 50px 0;
  width: 50%;
  margin: auto;
}

.home_font1 {
  font-size: 11px;
  color: #efefef9a;
}

.home_font2 {
  font-size: 50px;
  color: #efefef;
  padding: 20px 0;
}

.home_font3 {
  font-size: 11px;
  color: #efefef9a;
}

.home_section2 {
  width: 75%;
  margin: auto;
  padding: 20px 0 80px 0;
}

.home_section2 input {
  width: 100%;
  margin: auto;
  padding: 10px 10px;
  border-radius: 10px;
  outline: none;
  border: none;
}

.home_section2 input::placeholder {
  color: #172b4d;
  transition: 3s;
}

.home_section2 input:hover::placeholder {
  transform: translate(10px, 0);
  color: #172b4d50;
}

.home_section3 {
  width: 90%;
  margin: auto;
  padding: 40px;
  background-color: #020a11;
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  gap: 20px;
  border-radius: 15px;
}

.coin_article {
  background-color: white;
  align-items: center;
  display: flex;
  justify-content: center;
  padding: 10px 5px;
  gap: 10px;
  color: #111315;
  border-radius: 15px;
  transition: 1s;
}

.coin_article:hover {
  transform: scale(1.05);
}

.coin_img {
  width: 60px;
  object-position: center;
  object-fit: cover;
}

.coin_details_box {
  list-style: none;
  text-align: center;
}

.coin_details_item {
  padding: 5px 0;
  color: #020a11;
}

.home_section3_btn {
  padding: 8px 15px;
  background-color: #0077fe;
  border: none;
  border-radius: 10px;
}

@media only screen and (max-width: 1024px) {
  .home_font2 {
    font-size: 45px;
    color: #efefef;
    padding: 20px 0;
  }
}

@media only screen and (max-width: 820px) {
  .home_font1 {
    font-size: 9px;
    color: #efefef9a;
  }

  .home_font2 {
    font-size: 35px;
    color: #efefef;
    padding: 20px 0;
  }

  .home_font3 {
    font-size: 9px;
    color: #efefef9a;
  }

  .home_section3 {
    width: 90%;
    margin: auto;
    padding: 40px;
    background-color: #020a11;
    display: grid;
    grid-template-columns: repeat(3, 1fr);
    gap: 20px;
    border-radius: 15px;
  }
}

@media only screen and (max-width: 425px) {
  .home_section1 {
    padding: 30px 0 10px 0;
    width: 80%;
    margin: auto;
  }

  .home_font2 {
    font-size: 30px;
    color: #efefef;
    padding: 8px 0;
  }

  .home_font3 {
    display: none;
  }

  .home_section2 {
    width: 75%;
    margin: auto;
    padding: 0 0 20px 0;
  }

  .home_section2 input {
    width: 100%;
    margin: auto;
    padding: 5px 10px;
    border-radius: 10px;
    outline: none;
    border: none;
    font-size: 12px;
  }

  .home_section3 {
    width: 90%;
    margin: auto;
    padding: 10px 10px;
    background-color: #020a11;
    display: grid;
    grid-template-columns: repeat(2, 1fr);
    gap: 15px;
    border-radius: 15px;
  }

  .coin_article {
    background-color: white;
    align-items: center;
    display: flex;
    justify-content: center;
    padding: 3px 3px;
    gap: 10px;
    color: #111315;
    border-radius: 15px;
    transition: 1s;
    font-size: 11px;
  }

  .coin_img {
    width: 40px;
    object-position: center;
    object-fit: cover;
  }

  .home_section3_btn {
    display: none;
  }
}

@media only screen and (max-width: 375px) {
  .home_section1 {
    padding: 30px 0 20px 0;
    width: 90%;
    margin: auto;
  }
}
