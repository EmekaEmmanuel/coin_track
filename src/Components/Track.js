import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import { PropTypes } from 'prop-types';
import '../css/Track.css';

function Track(props) {
  const { myCoins } = props;
  const [inputCoin, setInputCoin] = useState('');

  const filteredCoins = myCoins.filter(
    (coin) => coin.name.toLowerCase().includes(inputCoin.toLowerCase())
      || coin.symbol.toLowerCase().includes(inputCoin.toLowerCase()),
  );

  return (
    <div className="app_container">
      <main className="main_tag">
        <section className="home_section1">
          <p className="home_font1">THE NEW STANDARD DECENTRALIZED TRACKING PLATFORM</p>
          <h1 className="home_font2">Track decentralized between blockchains</h1>
          <p className="home_font3">
            Swap all assets frictionless in a decentralized manner, earn yield on your native
            coins and manage
            your portfolio while you have full control of your keys.
          </p>
        </section>

        <section className="home_section2">
          <input type="text" placeholder="Enter Coin name or symbol" onChange={(e) => setInputCoin(e.target.value)} />
        </section>

        <section className="home_section3">

          {filteredCoins.map((coin) => (
            <article className="coin_article" key={coin.name}>
              <figure className="coin_figure">
                <img className="coin_img" src={coin.icon} alt="" />
              </figure>
              <div className="home_coin_details">
                <ul className="coin_details_box">
                  <li className="coin_details_item">
                    Rank:
                    {coin.rank}
                  </li>
                  <li className="coin_details_item">{coin.name}</li>
                  <li className="coin_details_item">{coin.symbol}</li>
                  <li className="coin_details_item">
                    $
                    {coin.price.toFixed(2)}
                  </li>
                  <button type="button" className="home_section3_btn">
                    <Link to={`/details/${coin.id}`} className="td_none">Track</Link>
                  </button>
                </ul>
              </div>
            </article>
          ))}
        </section>

      </main>
    </div>
  );
}

Track.propTypes = {
  myCoins: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.object)).isRequired,
};

export default Track;
