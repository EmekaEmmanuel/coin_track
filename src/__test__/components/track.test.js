import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import store from '../../redux/store';
import Track from '../../Components/Track';

it('Track component renders correctly', () => {
  const myCoins = store.getState().coin.coins;
  const tree = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <Track myCoins={myCoins} />
      </MemoryRouter>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
