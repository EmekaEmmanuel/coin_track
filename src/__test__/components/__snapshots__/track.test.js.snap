// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Track component renders correctly 1`] = `
<div
  className="app_container"
>
  <main
    className="main_tag"
  >
    <section
      className="home_section1"
    >
      <p
        className="home_font1"
      >
        THE NEW STANDARD DECENTRALIZED TRACKING PLATFORM
      </p>
      <h1
        className="home_font2"
      >
        Track decentralized between blockchains
      </h1>
      <p
        className="home_font3"
      >
        Swap all assets frictionless in a decentralized manner, earn yield on your native coins and manage your portfolio while you have full control of your keys.
      </p>
    </section>
    <section
      className="home_section2"
    >
      <input
        onChange={[Function]}
        placeholder="Enter Coin name or symbol"
        type="text"
      />
    </section>
    <section
      className="home_section3"
    />
  </main>
</div>
`;
